openai: 0.25.0 -> 0.26.1

malob opened the pull request:
###### Description of changes

* Updated to latest version (changes: https://github.com/openai/openai-python/compare/v0.25.0...v0.26.1)
* Updated `propagatedBuildInputs` and `passthru.optional-dependencies` to match changes made to package.
* Added `withOptionalDependencies` option.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

malob committed:
openai: 0.25.0 -> 0.26.1

Diff: https://github.com/openai/openai-python/compare/v0.25.0...v0.26.1

malob committed:
openai: add `withOptionalDependencies` option

malob commented:
Result of `nixpkgs-review pr 209934` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>openai</li>
    <li>python310Packages.openai</li>
    <li>python39Packages.openai</li>
  </ul>
</details>

malob commented:
Result of `nixpkgs-review pr 209934` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>openai</li>
    <li>python310Packages.openai</li>
    <li>python39Packages.openai</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/openai/default.nix:
```diff
@@ -18,9 +20,18 @@
 }:
 
+let
+  datalib = [
+    numpy
+    openpyxl
+    pandas
+    pandas-stubs
+  ];
```
Please inline this and use rec

malob replied:
Alrighty. (I thought it was best practice not to use `rec` if it was possible to do so without too much trouble.)

SuperSandro2000 commented on pkgs/development/python-modules/openai/default.nix:
```diff
@@ -29,29 +40,28 @@ buildPythonPackage rec {
-    wandb = [
-      wandb
-    ];
+    inherit datalib;
```
```suggestion
    datalib = [
      numpy
      openpyxl
      pandas
      pandas-stubs
    ];
```

SuperSandro2000 commented on pkgs/development/python-modules/openai/default.nix:
```diff
@@ -29,29 +40,28 @@ buildPythonPackage rec {
     typing-extensions
   ];
 
   passthru.optional-dependencies = {
```
```suggestion
  passthru.optional-dependencies = rec {
```

malob replied:
Can't use `rec` here cause it because `wandb` from `passthru.optional-dependencies` to shadow `wandb` (the Python package). So I used `passthru.optional-dependencies.datalib` since there is already a `rec` on for on the set of the package definition.

SuperSandro2000 commented on pkgs/development/python-modules/openai/default.nix:
```diff
@@ -29,29 +40,28 @@ buildPythonPackage rec {
     tqdm
+  ] ++ lib.optionals (pythonOlder "3.8") [
     typing-extensions
   ];
```
We can add the optional dependencies here.

malob replied:
Are you suggesting that we include all optional dependencies in `propagatedBuildInputs`?

I asked because in #194919 `passthru.optional-dependencies` was created. Before that change was made I used to just include all dependencies (optional or otherwise) in `propagatedBuildInputs`. I had assumed that change was made so that by default the packages would not include the optional dependencies, while giving folks an easy way to add the additional dependencies themselves if they wanted to include them. Which made sense to me.

I made the change to `openai` (defined in `all-packages.nix`) to include all the optional dependencies, so that folks using the Python package for the CLI utility would get all the functionality by default, while allowing folks who are using the Python package as a library to use `pythonXPackages.openai` and include only the optional dependencies they wanted/needed if any.

(Of course it may be that I'm confused or misunderstanding you, and therefore my comment doesn't make much sense. If that's the case, sorry about that.)

SuperSandro2000 replied:
> I asked because in https://github.com/NixOS/nixpkgs/pull/194919 passthru.optional-dependencies was created. Before that change was made I used to just include all dependencies (optional or otherwise) in propagatedBuildInputs. I had assumed that change was made so that by default the packages would not include the optional dependencies, while giving folks an easy way to add the additional dependencies themselves if they wanted to include them. Which made sense to me.

Yes, that's correct.



> I made the change to `openai` (defined in `all-packages.nix`) to include all the optional dependencies, so that folks using the Python package for the CLI utility would get all the functionality by default, while allowing folks who are using the Python package as a library to use `pythonXPackages.openai` and include only the optional dependencies they wanted/needed if any.

That wasn't obvious to me. I think in other packages we do a ``full`` variant. At least for me the current behavior would be a bit unexpected.

malob replied:
Done.

malob commented:
@SuperSandro2000, alrighty I've gone ahead and added a `-full` variant for the `openai` package. This now seems like the better course of action to me given that the optional dependencies are 1+ GB in size. 

Also, having now rebased on `master`, `openai-full` now fails to build due to one of the optional dependencies, `wandb`, having build issues. This seems like another good argument for your preference for a `-full` variant, as opposed to `openai` including optional dependencies by default, since when this kind of thing happens folks will still be able to use `openai` which is much less likely to have build issues due to not depending on many dependencies in `python3Packages` that often run into build issues.

malob commented:
Result of `nixpkgs-review pr 209934` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>openai-full</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openai (python310Packages.openai)</li>
    <li>python311Packages.openai</li>
  </ul>
</details>

malob commented:
Result of `nixpkgs-review pr 209934` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>openai-full</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openai (python310Packages.openai)</li>
    <li>python311Packages.openai</li>
  </ul>
</details>

SuperSandro2000 commented:
https://hydra.nixos.org/build/205493722 

Going to ask on matrix with that

malob commented:
Removed `openai-full` for now since it doesn't currently build due to issues with `wandb`.

malob commented:
Result of `nixpkgs-review pr 209934` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openai (python310Packages.openai)</li>
    <li>python311Packages.openai</li>
  </ul>
</details>

malob commented:
Result of `nixpkgs-review pr 209934` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openai (python310Packages.openai)</li>
    <li>python311Packages.openai</li>
  </ul>
</details>
